File: golang/283. Move Zeroes .go
func moveZeroes(nums []int)  {
    j := 0;
    
    for i, e := range nums {
        if e != 0 {
            nums[j] = nums[i] 
            j++
        }
    }
    
    for ; j < len(nums); j++ {
        nums[j] = 0
    } 
}

File: java/283. Move Zeroes .java
class Solution {
    public void moveZeroes(int[] nums) {
        
        Queue<Integer> nonzeros = new LinkedList<>();
        
        for(int num : nums)
            if(num != 0)
                nonzeros.add(num);
        
        Arrays.fill(nums, 0);
        
        for(int i = 0; !nonzeros.isEmpty(); i++)
            nums[i] = nonzeros.poll();
    }
}
